### PR TITLE
Use kernel_flavor when polling gitbuilders

### PR DESCRIPTION
**Pull request by zmc**
http://tracker.ceph.com/issues/11645

Signed-off-by: Zack Cerza <zack@redhat.com>

### CODE REVIEW

**Comment by zmc** [comment]
This is tested, but we may want to hold off on merging until Monday

**Comment by andrewschoen** [comment]
This all looks pretty sane to me.  Waiting until Monday is fine as well.

**Comment by zmc** [comment]
I *think* I'm happy with this, but I should probably test it via a scheduled run. Not sure this moment how that should best be done.

File: teuthology/suite.py [hunk 1]
```diff
@@ -594,10 +617,29 @@ def schedule_suite(job_config,
     return count
 
 
-def get_package_versions(sha1, os_type, package_versions=None):
+def get_install_task_flavor(job_config):
```
Comment (andrewschoen) [inline teuthology/suite.py]: Tests on this method would be nice.

File: teuthology/task/install.py [hunk 2]
```diff
@@ -1222,6 +1222,31 @@ def ship_utilities(ctx, config):
         )
 
 
+def get_flavor(config):
```
Comment (andrewschoen) [inline teuthology/task/install.py]: Tests would also be nice here.

Comment (zmc) [inline teuthology/task/install.py]: Added!

File: teuthology/suite.py (hunk 1 above)
Comment (zmc) [inline teuthology/suite.py]: Added!

**Comment by zmc** [comment]
So this should be ready to go.

I observed an issue where it wasn't doing what I wanted here:
http://qa-proxy.ceph.com/teuthology/zack-2015-05-20_15:39:30-fs:verify-master---basic-vps/902151/teuthology.log

But! That's only happening because of the weird vps-specific distro defaults in ``misc.get_distro_version()``. I want to fix that and probably consolidate it with ``suite.get_distro_defaults()``, but I think that's outside the (already large) scope of this PR.

**Pull request merged**